utils/src/activityLog/Activity.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { randomUUID } from "crypto";
+import { EventEmitter, TreeItemCollapsibleState } from "vscode";
+import * as types from '../../index';
+import { parseError } from "../parseError";
+import { AzExtTreeItem } from "../tree/AzExtTreeItem"
+
+export interface ActivityTreeItemOptions {
+    label: string;
+    contextValuePostfix?: string;
+    collapsibleState?: TreeItemCollapsibleState;
+    children?: (parent: types.AzExtParentTreeItem) => AzExtTreeItem[];
+}
+
+export abstract class ActivityBase implements types.Activity {
+
+    public readonly onStart: typeof this._onStartEmitter.event;
```

Comment:
This is valid, but another way that saves a few lines here and in the constructor is to init like this:
```typescript
private readonly _onStartEmitter = new EventEmitter<types.OnStartActivityData>();
// ...
private readonly onStart = this._onStartEmitter.event;
```
I don't feel strongly one way or the other so do whichever you prefer :smile: